db/migrate/20151110191852_create_redirections.rb

```diff
@@ -0,0 +1,10 @@
+class CreateRedirections < ActiveRecord::Migration
+  def change
+    create_table :redirections do |t|
+      t.timestamps null: false
+      t.references :next
```

Comment:
I like DB validations for uniqueness because they actually provide a service: AR uniqueness validations can have race conditions.